codalab/server/bundle_manager.py

```diff
@@ -436,11 +430,21 @@ def get_sort_key(worker):
             else:
                 deps = set(worker['dependencies'])
                 num_available_deps = len(needed_deps & deps)
-            worker_id = worker['worker_id']
 
-            # if the bundle doesn't request GPUs (only request CPUs), prioritize workers that don't have GPUs
-            gpu_priority = bundle_resources.gpus or not has_gpu[worker_id]
-            return (gpu_priority, num_available_deps, worker['cpus'], random.random())
+            # If the bundle requested for GPUs, prioritize only those workers that have GPUs.
+            # If the bundle didn't request for GPUs (a CPU only job), prioritize all workers based criteria as follows:
+            # 1. number of running jobs.
+            # 2. number of available dependencies.
+            # 3. number of CPUs.
+            # 4. random seeds.
+            gpu_priority = bundle_resources.gpus and worker['gpus']
```

Comment:
`and`?

Reply:
Yes. `1 and 1` returns `1`. `0 and 1` returns `0`.

Reply:
Oh, i see. this won't work when number of `gpus > 1`